### ISSUE TITLE
build -i doesn't accept multiple arguments since 0.12.7

### ISSUE THREAD

**Issue by hombit** [PyO3/maturin#872]
### Bug Description

Parsing of an arguments list doesn't supported since 0.12.7, but it looks like a breaking change. I also have found no information about this change in the changelog
```sh
# maturin build -i python3.7 python3.8 python3.9 python3.10
error: Found argument 'python3.8' which wasn't expected, or isn't valid in this context

USAGE:
    maturin build [OPTIONS]

For more information try --help
```

`maturin build -i python3.7 -i python3.8` works as expected

### Your Python version (`python -V`)

python 3.9.9

### Your pip version (`pip -V`)

21.3.1

### What bindings you're using

pyo3

### Does `cargo build` work?

- [ ] Yes, it works

### If on windows, have you checked that you aren't accidentally using unix path (those with the forward slash `/`)?

- [ ] Yes

### Steps to Reproduce

1. Install maturin >= 0.12.7
2. `maturin build -i python3.7 python3.8`

**Comment by messense** [comment]
I think it's a side effect of [upgrading to clap 3.0](https://github.com/PyO3/maturin/pull/738). Clap 3.0 has splitted the meaning of `multiple` to `multiple_values` and `multiple_occurrences` and it recommended to use `multiple_occurrences` when upgrading. See https://docs.rs/clap/latest/clap/struct.Arg.html#method.multiple_values

We can certainly bring multiple value support back, but it does leave a question how do we deal with other options that can appear multiple times?

**Comment by hombit** [comment]
I'm not sure if it is not too late to fix it in 0.12.x, but fixing Changelog and readme could help other people
